### PR TITLE
Add disable methods

### LINKED ISSUE #11864: RFC: Split methods of type `enableFoo()` with boolean argument.
While updating my plugins for CakePHP 3.6 I felt odd having to do method calls like `$query->enableHydration(false)`.

It would be better to separate methods `enableFoo()` for enabling a flag and `disableFoo()` for disabling a flag.

These are the enabling methods which take boolean argument which would have to be changed:

- `\Cake\ORM\Query::enableHydration(bool $enable = true)`
- `\Cake\Database\Connection::enableSavePoints(bool $enable)` // This odd one doesn't have a default value
- `\Cake\Database\Driver::enableAutoQuoting(bool $enable = true)`
- `\Cake\Database\Query::enableBufferedResults(bool $enable = true)`
- `\Cake\View\ViewBuilder::enableAutoLayout(bool $enable = true)`

### PR DESCRIPTION
**Pull request by ADmad**
Closes #11864.

### CODE REVIEW

File: src/ORM/Query.php [hunk 1]
```diff
@@ -1010,6 +1010,22 @@ public function enableHydration($enable = true)
...
+     * Disable hydrating entities.
+     *
+     * Disabling hydration will cause array results to be returned for the query
+     * instead of entities.
+     *
+     * @return $this
+     */
+    public function disableHydration()
+    {
+        $this->_dirty();
```
Comment (dereuromark) [inline src/ORM/Query.php]: Instead of duplicating the logic this must call the actual wrapper internally to prevent unDRY and copy paste bugs.

Comment (ADmad) [reply src/ORM/Query.php]: I would rather have a couple of line of code duplication then adding additional method call.

Comment (dereuromark) [reply src/ORM/Query.php]: Dont nano opt a syntactic sugar wrapper. Thats a nogo, the wrapping is cheap and will prevent future bugs.

**Review by markstory — APPROVED** [review]
Looks good to me.

**Comment by dereuromark** [comment]
:cry:

**Pull request merged**